object-api-multipart.go

```diff
@@ -106,51 +107,52 @@ func (o objectAPI) listMetaVolumeFiles(prefixPath string, markerPath string, rec
 		// Loop through and validate individual file.
 		for _, fi := range fileInfos {
 			var entries []FileInfo
-			// List all the entries if fi.Name is a leaf directory, if
-			// fi.Name is not a leaf directory them the resulting
-			// entries are empty.
-			entries, e = o.listLeafEntries(fi.Name)
-			if e != nil {
-				log.WithFields(logrus.Fields{
-					"prefixPath": fi.Name,
-				}).Errorf("%s", e)
-				return nil, false, e
+			if fi.Mode.IsDir() {
+				// List all the entries if fi.Name is a leaf directory, if
+				// fi.Name is not a leaf directory them the resulting
+				// entries are empty.
+				entries, e = o.listLeafEntries(fi.Name)
+				if e != nil {
+					log.WithFields(logrus.Fields{
+						"prefixPath": fi.Name,
+					}).Errorf("%s", e)
+					return nil, false, e
+				}
 			}
 			// Set markerPath for next batch of listing.
 			markerPath = fi.Name
 			if len(entries) > 0 {
+
```

Comment:
> Then better place is before the for loop.

no, that will be out of context

```
                        if len(entries) > 0 {                                                                                                                                     
                                                                                                                                                                                  
                                // We reach here for non-recursive case and a leaf entry.                                                                                         
                                                                                                                                                                                  
                                for _, entry := range entries {                                                                                                                   
                                        allFileInfos = append(allFileInfos, entry)                                                                                                
                                        newMaxKeys++                                                                                                                              
                                        if newMaxKeys == maxKeys {                                                                                                                
                                                // eof value is automatically taken care of.                                                                                      
                                                return                                                                                                                            
                                        }                                                                                                                                         
                                }                                                                                                                                                 
                                continue                                                                                                                                          
                        }                       
```

see, it's a code flow comment inside the if condition